### PR TITLE
Feature/s3 sync efs

### PR DESCRIPTION
**Pull request by MartinXPN**

### CODE REVIEW

File: sync/app.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    with TemporaryDirectory() as extraction_dir, ZipFile(zip_path, 'r') as zip_ref:
+        zip_ref.extractall(extraction_dir)
+        targets = glob.glob(f'{extraction_dir}/**/*.ans.txt', recursive=True) + \
+                  glob.glob(f'{extraction_dir}/**/*.out.txt', recursive=True) + \
+                  glob.glob(f'{extraction_dir}/**/*.a', recursive=True) + \
+                  glob.glob(f'{extraction_dir}/**/*.ans', recursive=True) + \
+                  glob.glob(f'{extraction_dir}/**/*.out', recursive=True)
+        targets = sorted(targets)
+        print('targets:', targets)
+        inputs = [t.replace('.a', '') if t.endswith('a') else t.replace('ans', 'in') for t in targets]
```
Comment (edogrigqv2) [inline sync/app.py:41]: Logic here should be different. 
Remove the .txt suffix if it exists. 
Then if it ends with .a, .out or .ans remove that suffix and replace it with empty string or '.in'
Probably create a separate function for this. That function can be unit tested as well.

File: sync/app.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    print('encryption key len:', len(encryption_key))
+    fernet = Fernet(encryption_key)
+
+    # Compress:   (1) json.dumps   (2) .encode('utf-8')   (3) gzip.compress()   (4) encrypt
+    # Decompress: (1) decrypt      (2) gzip.decompress()  (3) .decode('utf-8')  (4) json.loads()
+    tests = json.dumps(tests)                                               # (1)
+    print('initial sys.size of tests:', sys.getsizeof(tests))
+    big = sys.getsizeof(tests) > 50 * 1024 * 1024
+    tests = tests.encode('utf-8')                                           # (2)
+    tests = gzip.compress(tests, compresslevel=9 if not big else 7)         # (3)
```
Comment (edogrigqv2) [inline sync/app.py:65]: Why not compress everything with level 9?
Lambdas are super cheap anyway.

**Review by edogrigqv2 — CHANGES_REQUESTED** [review]

File: sync/app.py (hunk 2 above)
Comment (MartinXPN) [reply sync/app.py:65]: level 9 for files >= 50MB can take 30 minutes or more. 
The compression differences between level 9 and level 7 are not that different (in terms of size) but are really different in terms of time

File: sync/app.py (hunk 1 above)
Comment (MartinXPN) [reply sync/app.py:41]: didn't quite understand what's the advantage of removing suffixes instead of including all?

**Pull request merged**